huaweicloud/services/dms/resource_huaweicloud_dms_kafka_instance.go

```diff
@@ -321,11 +325,85 @@ func getKafkaProductDetail(config *config.Config, d *schema.ResourceData) (*prod
 	return nil, fmtp.Errorf("can not found product detail base on product_id: %s", productID)
 }
 
+func updateCrossVpcAccesses(client *golangsdk.ServiceClient, d *schema.ResourceData) error {
+	oldVal, newVal := d.GetChange("cross_vpc_accesses")
+	var crossVpcAccess []map[string]interface{}
+	if len(oldVal.([]interface{})) < 1 {
+		v, err := instances.Get(client, d.Id()).Extract()
+		if err != nil {
+			return fmtp.Errorf("error getting DMS instance: %v", err)
+		}
+		crossVpcAccess, err = flattenConnectPorts(v.CrossVpcInfo)
+		if err != nil {
+			return fmtp.Errorf("[ERROR] error retriving details of the cross-VPC information: %v", err)
+		}
+	} else {
+		oldAccesses := oldVal.([]interface{})
+		for _, val := range oldAccesses {
+			crossVpcAccess = append(crossVpcAccess, val.(map[string]interface{}))
+		}
+	}
+
+	newAccesses := newVal.([]interface{})
+	contentMap := make(map[string]string)
+	for i, oldAccess := range crossVpcAccess {
+		lisIp := oldAccess["lisenter_ip"].(string)
+		// If we configure the advertised ip as ["192.168.0.19", "192.168.0.8"], the length of new accesses is 2, and
+		// the length of old accesses is always 3.
+		if len(newAccesses) > i {
+			// Make sure the index is valid.
+			newAccess := newAccesses[i].(map[string]interface{})
+			// Since the "advertised_ip" is already a definition in the schema, the key name must exist.
+			if advIp, ok := newAccess["advertised_ip"].(string); ok && advIp != "" {
+				contentMap[lisIp] = advIp
+				continue
+			}
+		}
+		contentMap[lisIp] = lisIp
+	}
+
+	opts := instances.CrossVpcUpdateOpts{
+		Contents: contentMap,
+	}
+	result, err := instances.UpdateCrossVpc(client, d.Id(), opts)
```

Comment:
please check the API is available in all regions of huaweicloud and joined-operation clouds.